### DISCUSSION TITLE
Remove UniqueLocks and UpgradeToUnique

### DISCUSSION THREAD

**Discussion by bpmckinnon** [greg7mdp/parallel-hashmap#220] [General]
I was looking at the new ReadWriteLock change and I'm wondering if UniqueLocks and UpgradeToUnique should be removed to enforce a single mechanism for upgrading locks.  Is there any plan to remove it?

**Comment by greg7mdp** [comment]
Yes, I think I should remove `UpgradeToUnique` and update the one location where it is used, thanks for the reminder. I'll have to think some more about `UniqueLocks`.

**Comment by greg7mdp** [comment]
`UniqueLocks` is still used in merge (see [this](https://github.com/greg7mdp/parallel-hashmap/blob/9248dcf07ec5c0499e6cc7e7555a746c58130934/parallel_hashmap/phmap.h#L3526)). What do you think of this PR https://github.com/greg7mdp/parallel-hashmap/pull/221 for removing the unused `Upgrade` stuff?
